server/server.go

```diff
@@ -75,33 +78,30 @@ func (s *Server) RunUntilShutdown() error {
 	// the interrupt handler.
 	<-s.interceptor.ShutdownChannel()
 
+	// shutdown servers
+	s.grpcServer.GracefulStop()
+	s.httpServer.Shutdown(context.Background())
```

Comment:
what's the reason to move the shutdown handling to RunUntilShutdown?

also there is lint err here

Reply:
single place to handle shutdown, in each server, there is the potential race condition on same shutdown

lint err should be fixed

Reply:
i think it makes sense to have a single shutdown point. but what's the race condition?

Reply:
i am not sure if they will race the ShutdownChannel between two servers in each server start function